### PR TITLE
Fix for infinite loop issue reported in #1009

### PR DESCRIPTION
**Pull request by blickly**
As a first step to fixing https://github.com/angular/tsickle/issues/1009, I've tried to reproduce the issue in a unit test.

Surprisingly, it already passes.

### CODE REVIEW

File: test/tsickle_test.ts [hunk 1]
```diff
@@ -105,6 +105,21 @@ describe('emitWithTsickle', () => {
...
+    };
+
+    const jsSources = emitWithTsickle(
+        tsSources, {
+          preserveConstEnums: true,
+          module: ts.ModuleKind.ES2015,
+        },
+        {es5Mode: false, googmodule: false});
+
+      expect(jsSources['a.js']).toContain(`export function f() { return f; }`);
```
Comment (evmar) [inline test/tsickle_test.ts]: I would expect this output to contain some Closure stuff in it (e.g. "exports.f = ..."), is it possible this isn't running the right emit?

Comment (blickly) [reply test/tsickle_test.ts]: The full emit that I saw was:
```ts
/**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /**
     * @return {?}
     */
    export function f() { return f; }
``

So it looks like it is producing the closure type annotations (which I has assumed would be enough to trigger the bug).  Do you think trying goog.module emit would be needed to trigger the bug?

Comment (evmar) [reply test/tsickle_test.ts]: I suspect this:
https://github.com/angular/tsickle/blob/3276f179c3d52ab5a7d3f2d88f57e92797f2bd1b/test/tsickle_test.ts#L37

This is the flag that we flipped in google but forgot about in this test suite.  :(

I think we should drop the flag entirely, but try flipping it here for now to see if that unsticks you.

Comment (blickly) [reply test/tsickle_test.ts]: Aha, that was it. Thanks Evan!

**Comment by blickly** [comment]
OK, this looks good to go now

File: src/type_translator.ts [hunk 2]
```diff
@@ -624,6 +624,7 @@ export class TypeTranslator {
       }
       return typeStr;
     } else if (type.objectFlags & ts.ObjectFlags.Anonymous) {
+      this.seenAnonymousTypes.add(type);
```
Comment (evmar) [inline src/type_translator.ts]: It feels like this belongs in translateAnonymousType.  Maybe the rest of the code in this block (before the call to this.translateAnonymousType) belongs in translateAnonymousType as well?

Comment (blickly) [reply src/type_translator.ts]: Fair enough.  After doing so, it made it more obvious that type.symbol was being checked redundantly, so I simplified slightly.

Comment (evmar) [reply src/type_translator.ts]: Looks great.

**Comment by evmar** [comment]
I think the CI is failing because you need to run the formatter (it's printed in the CI log)

**Comment by blickly** [comment]
Yes, that was exactly it.  Thanks!

**Comment by evmar** [comment]
What are the chances we need to TGP this first?  I always think "what could possibly go wrong" and then something inevitably goes horribly wrong...

**Pull request merged**